### ISSUE TITLE
Roborock S5 - no persistent data/lab mode

### ISSUE THREAD

**Issue by mtjoernelund** [rand256/valetudo#259]
Just updated from 9.4, firmware version 2008 to the latest binary available here, i.e. 0.9.5.2 with firmware version 2020. Updated via web interface and update seem to have gone fine, save that persistent data/lab mode no longer is available.

Settings - > System Settings states that “Sorry, only Roborock S5x supports the persistent map feature”

Under Zones I no longer have the possibility of editing and renaming rooms.

However, the previous map with edited rooms (but not their names) still shows.

Vacuum Model: Gen2/S5 

Valetudo Version: 0.9.5.2

**Comment by rand256** [comment]
Just reopen the page. It happens sometimes after a reboot.

Though I'll see what can I do to prevent this from happening.

**Comment by mtjoernelund** [comment]
Thought I had tried reloading the page and rebooting, without success. However after a cleaning run and return to dock it seem to have fixed itself. Not sure if it was just another reload of the page, the cleaning run, returning to dock or some combination that fixed it. 

Thank you for your great work on this.

**Comment by rand256** [comment]
Web UI claims you have no persistent data/lab mode only because it was unable to get current_status from API on the first launch (most likely API simply wasn't ready after the reboot at that point), and by default it falls back to gen1 device in this case. Doing a cleaning run can't fix it since required variables are set only when you (re-)open the UI.

tl;dr: this needs to be improved one or other way.

**Comment by mtjoernelund** [comment]
Noted, and thanks for the explanation.
